src/mito2/src/memtable/key_values.rs

```diff
@@ -143,9 +166,37 @@ pub struct KeyValue<'a> {
     helper: &'a SparseReadRowHelper,
     sequence: SequenceNumber,
     op_type: OpType,
+    primary_key_encoding: PrimaryKeyEncoding,
 }
 
 impl KeyValue<'_> {
+    /// Returns primary key encoding.
+    pub fn primary_key_encoding(&self) -> PrimaryKeyEncoding {
+        self.primary_key_encoding
+    }
+
+    /// Returns the partition key.
+    pub fn partition_key(&self) -> u32 {
+        if self.primary_key_encoding == PrimaryKeyEncoding::Sparse {
+            let Some(primary_key) = self.primary_keys().next() else {
+                return 0;
+            };
+            let key = primary_key.as_binary().unwrap().unwrap();
+
+            let mut deserializer = Deserializer::new(key);
+            deserializer.advance(COLUMN_ID_ENCODE_SIZE);
+            let field = SortField::new(ConcreteDataType::uint32_datatype());
+            let table_id = field.deserialize(&mut deserializer).unwrap();
+            table_id.as_value_ref().as_u32().unwrap().unwrap()
```

Comment:
Well, I'm going to refactor the memtable to avoid this hack......

Reply:
Left a TODO for you